Apache-Beam-Jobs.md

```diff
@@ -1,55 +1,85 @@
+* [Running Apache Beam Jobs](#running-apache-beam-jobs)
+* [Apache Beam Job Architecture](#apache-beam-job-architecture)
+  * [`Pipeline`s](#pipelines)
+  * [`PValue`s](#pvalues)
+  * [`PTransform`s](#ptransforms)
+    * [`ParDo` and `DoFn`](#pardo-and-dofn)
+    * [`Map` and `FlatMap`](#map-and-flatmap)
+    * [`Filter`](#filter)
+    * [`GroupByKey`](#groupbykey)
+    * [Example of using `GroupByKey`,`Filter`, and `FlatMap`](#example-of-using-groupbykeyfilter-and-flatmap)
+  * [`Runner`s](#runners)
+  * [High-level Guidelines](#high-level-guidelines)
+* [Case studies](#case-studies)
+  * [Case study: `CountAllModelsJob`](#case-study-countallmodelsjob)
+  * [Case Study: `SchemaMigrationJob`](#case-study-schemamigrationjob)
 
--   **Continuous**: Operations that are designed to run _indefinitely_ by
-    "reacting to updates" in the datastore.
+## Introduction
 
-    Examples:
-    -   Updating the top `N` answers to a lesson
-    -   Generating notifications for the events that users have subscribed to
+[Apache Beam](https://beam.apache.org/) is used by Oppia to perform large-scale datastore operations. There are two types of operations:
 
-If you're already familiar with Apache Beam, or are eager to start writing a new
-job, then follow the [Quick Start](#quick-start).
+* **Batch**: Operations that are designed to be executed _once_ on the current state of the datastore. Here are some examples:
 
-Otherwise, if you're looking for supplemental material, you can read the entire
-guide and fallback to the [Apache Beam Programming Guide][1] for more details.
+  * Count the number of models in the datastore.
+  * Update the StringProperty values of all models with a specific kind.
+  * Validate the relationships between models.
 
-Running Apache Beam Jobs
-========================
+* **Continuous**: Operations that are designed to run _indefinitely_ by reacting to updates to the datastore. Here are some examples:
 
-Developer Environment
----------------------
-![screencast of running jobs](https://user-images.githubusercontent.com/5094060/128743997-70cca5f9-0b76-4294-806e-f65f5df5be95.gif)
-1. Sign in as an administrator ([instructions][3])
-2. Navigate to **Admin Page > Roles Tab**
-3. Add the "Release Coordinator" role to the username you are signed in with
-4. Navigate to http://localhost:8181/release-coordinator, then to the **Beam Jobs tab**
-5. Search for your job and then click the **Play button**
+  * Updating the top 10 answers to a lesson every time a new answer is submitted.
+  * Generating notifications for the events that users have subscribed to whenever those events change.
 
-Apache Beam Job Architecture
-============================
+If you're already familiar with Apache Beam, or are eager to start writing a new job, then jump to the [Quick Start](#quick-start). Otherwise, you can read the whole page. If you still have questions after reading, take a look at the [Apache Beam Programming Guide][1] for more details.
```

Comment:
```suggestion
If you're already familiar with Apache Beam or are eager to start writing a new job,  jump to the [case studies](#case-studies). Otherwise, you can read the whole page. If you still have questions after reading, take a look at the [Apache Beam Programming Guide][1] for more details.
```

Reply:
done